src/loaders/TiffLoader.ts

```diff
@@ -20,32 +21,48 @@ function prepareXML(xml: string): string {
 
 function getOME(xml: string): Element {
   const parser = new DOMParser();
-  const xmlDoc = parser.parseFromString(xml, "text/xml");
-  const omeEl = xmlDoc.getElementsByTagName("OME")[0];
-  return omeEl;
+  try {
+    const xmlDoc = parser.parseFromString(xml, "text/xml");
+    return xmlDoc.getElementsByTagName("OME")[0];
+  } catch (e) {
+    throw new VolumeLoadError("Could not find OME metadata in TIFF file", {
+      type: VolumeLoadErrorType.INVALID_METADATA,
+      cause: e,
+    });
+  }
 }
 
 class OMEDims {
   sizex = 0;
   sizey = 0;
-  sizez = 0;
-  sizec = 0;
-  sizet = 0;
+  sizez = 1;
+  sizec = 1;
+  sizet = 1;
   unit = "";
   pixeltype = "";
   dimensionorder = "";
-  pixelsizex = 0;
-  pixelsizey = 0;
-  pixelsizez = 0;
+  pixelsizex = 1;
+  pixelsizey = 1;
+  pixelsizez = 1;
   channelnames: string[] = [];
 }
 
+function getAttributeOrError(el: Element, attr: string): string {
+  const val = el.getAttribute(attr);
+  if (val === null) {
+    throw new VolumeLoadError(`Missing attribute ${attr} in OME-TIFF metadata`, {
```

Comment:
There's kind of both a principled and a practical explanation for what `wrapVolumeLoadError` does.

The principled version is that `wrapVolumeLoadError` is for responding to someone else's error. I.e., when something we do causes a library or the runtime to throw an error, we should catch it, put it in the `cause` field of a new `VolumeLoadError`, and re-throw it with a message that describes what we were trying to do in terms of the loader's lifecycle.

The practical version (why it actually exists) is because using it inside a `catch` call was the nicest way I could find to make error handling work through promises. It felt cleaner to me to do this kind of compositional error handling like this:

```typescript
const result = await doSomethingAsync().catch(wrapVolumeLoadError("Error description", ERROR_TYPE));
```

...rather than like this:

```typescript
let result: ReturnType<doSomethingAsync>;
try {
    result = await doSomethingAsync();
} catch (e) {
    throw new VolumeLoadError("Error description", { type: ERROR_TYPE, cause: e });
}
```